helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -539,7 +539,11 @@ private void subscribeChildChange(String path, NotificationContext.Type callback
         logger.debug(_manager.getInstanceName() + " subscribes child-change. path: " + path
             + ", listener: " + _listener);
       }
-      _zkClient.subscribeChildChanges(path, this);
+      if (callbackType == Type.INIT) {
```

Comment:
See watchForchilds comment.

We discussed this in the meeting.

The second case is actually needed for current state callbackhandler registration. LiveInstance change would cause current state callbackhandler to be added to ZkHelixManager. And at the time, it can often be the case the parent node of current state is not yet created. Note, this is the INIT case.

Later CALLBACK case, we should expect the current state parent path be there. Otherwise, we should not register the watch to Zk server (or we will leak a path). That is why we need the first case.

Reply:
// add a comment why we want to this